packages/core/src/routes/interaction/types/guard.ts

```diff
@@ -62,10 +62,10 @@ const phoneProfileGuard = phonePasscodePayloadGuard.pick({ phone: true });
 const socialProfileGuard = socialConnectorPayloadGuard.pick({ connectorId: true });
 
 export const registerProfileSafeGuard = z.union([
-  usernamePasswordPayloadGuard,
-  emailProfileGuard,
-  phoneProfileGuard,
-  socialProfileGuard,
+  usernamePasswordPayloadGuard.merge(profileGuard.omit({ username: true, password: true })),
+  emailProfileGuard.merge(profileGuard.omit({ email: true })),
+  phoneProfileGuard.merge(profileGuard.omit({ phone: true })),
+  socialProfileGuard.merge(profileGuard.omit({ connectorId: true })),
```

Comment:
looks wield, seems like they are equal to `profileGuard`?

Reply:
profileGuard all the fields are optional. For the register flow, we need to make sure at least one essential profile requirement is met. 
e.g. 
- username + password,  email, or phone is optional in this case
- If the email is provided,  all the rest fields will be optional. 
- but a single password or a single username should throw